bitrise/util.go

```diff
@@ -458,8 +458,14 @@ func removeStepDefaultsAndFillStepOutputs(stepListItem *models.StepListItemModel
 			return err
 		}
 
-		if err := repo.CloneTagOrBranch(stepIDData.IDorURI, stepIDData.Version).Run(); err != nil {
-			return err
+		if stepIDData.Version == "" {
+			if err := repo.Clone(stepIDData.IDorURI).Run(); err != nil {
```

Comment:
`repo.Clone` is not using the `--recursive` flag (which seems to be `--recurse-submodules` in recent git versions), dropping this flag is intentional?

Reply:
Not intentional, no, I didn't notice it. That code is in `go-utils`, so we'd have to update it there. Now that you point it out, it seems odd that the `CloneTagOrBranch` method clones recursively at all.

I'm not sure how serious of an issue it would cause to omit the flag.

I can see a few options:

1. Leave it as is.
2. Add a call to `SubmoduleUpdate` after the call to `Clone`.
3. Add the `--recursive` flag to the `Clone` method in `go-utils`.
4. Add a method to `go-utils` called `CloneRecursively` or something, and call that instead of `Clone` here.
5. Remove the `--recursive` flag from `CloneTagOrBranch` in `go-utils`, and add a call to `SubmoduleUpdate` after the call to `Clone` and `CloneTagOrBranch`.

I see pros and cons to each, but I think `#5` seems the _most_ correct to me. I don't think the `--recursive` flag should be there in my opinion. We'd have to update `go-utils` and bump the version though as it would be a breaking change.

wdyt?

Reply:
The v1 git package is quite inflexible, and we are anyway a matter of migrating to v2, where we will have the opportunity to design a way better package.

Since branch or tag cloning can be done with a single function call, I would keep it for the default branch cloning too.
WDYT of a new function, like this:

```
func (g *Git) CloneWithOpts(repo string, opts ...string) *command.Model {
	args := []string{"clone"}
	args = append(args, opts...)
	args = append(args, repo, ".")
	return g.command("clone", repo, ".")
}
```

(`Git.Clone` and `Git.CloneTagOrBranch` could call this new `Git.CloneWithOpts` to avoid code duplication)